mlir/lib/Bindings/Python/IRCore.cpp

```diff
@@ -3718,6 +3724,36 @@ void mlir::python::populateIRCore(py::module &m) {
             mlirValueReplaceAllUsesOfWith(self.get(), with.get());
           },
           kValueReplaceAllUsesWithDocstring)
+      .def(
+          "replace_all_uses_except",
+          [](PyValue &self, PyValue &with, PyOperation &exception) {
```

Comment:
It should be possible to take `MlirValue` here directly and avoid the conversion below.